### ISSUE TITLE
Log reasons why Type.GetType fails when trying to map enclosed message type string to CLR type

### ISSUE THREAD

**Issue by FMartens-ATS** [Particular/NServiceBus#6017]
Please have a look at the GetType method at:
 https://github.com/Particular/NServiceBus/blob/master/src/NServiceBus.Core/Unicast/Messages/MessageMetadataRegistry.cs

```
  Type GetType(string messageTypeIdentifier)
  {
      if (!cachedTypes.TryGetValue(messageTypeIdentifier, out var type))
      {
          type = Type.GetType(messageTypeIdentifier, false);
          cachedTypes[messageTypeIdentifier] = type;
      }

      return type;
  }
```
Type.GetType may return 'null' eventually leading to the following exception:
`System.InvalidOperationException: No handlers could be found for message type: XxXxXx.YyY.ZzZ at NServiceBus.LoadHandlersConnector.<Invoke>d__1.MoveNext() ` when there is no handler for the other message types in the EnclosedMessageTypes header parameter.

Would it be possible to log an error or other message informing the user that something went wrong retrieving the type?

**Comment by andreasohlund** [comment]
Looking at the code we do log at DEBUG https://github.com/Particular/NServiceBus/blob/master/src/NServiceBus.Core/Unicast/Messages/MessageMetadataRegistry.cs#L55 for each attempt to process the message but you are saying that we should log that higher (I think a WARN would be appropriate) the first time we detect `GetType` returning null?

**Comment by FMartens-ATS** [comment]
My apologies, let me rephrase. Would it be possible to log why Type.GetType failed?

**Comment by andreasohlund** [comment]
>My apologies, let me rephrase. Would it be possible to log why Type.GetType failed?

Yea, that sounds reasonable and was what I had in mind for that WARN that I mentioned above. 

Would you want to retitle this issue to something like "Log reasons why Type.GetType fails when trying to map enclosed message type string to CLR type" or something along those lines?

**Comment by FMartens-ATS** [comment]
Our problem was that our contracts assembly was not 'properly' loaded. The NServiceBus code was the first trying to get a type from our assembly which failed (File IO exception). I noticed the 'No handler for...' message in ServiceInsight and later on the other message you referred to. Both log entries did not tell me what the actual problem was so I modified the GetType method to be able to log the exception message and that lead me to the solution (ensuring that the assembly is properly loaded).

**Comment by andreasohlund** [comment]
Got it!

Do you agree that WARN is an appropriate level or should we go higher?

**Comment by FMartens-ATS** [comment]
A warning is more than enough. Thanks for picking this up.

**Comment by andreasohlund** [comment]
I've raised https://github.com/Particular/NServiceBus/issues/6017 to address this
